proposals/1412/client_proxy.md

```diff
@@ -0,0 +1,161 @@
+# Problem Statements
+Currently, OpenLineage integrations are directly sending events to the configured backend, which is not flexible enough for some situations such as
+- Not responsive backend due to an outage or a network issue
+- Long latency of the API that adds extra delay to the processing
+
```

Comment:
Proposal precisely defines a problem and presents a solution to it. My concern is that a problem definition is kind of narrowed while the ultimate goal should be `Propose architecture that allows a guarantee to users that no events are lost` which strictly relates to high availability (HA) of a system. The problem here is that storing data locally on disks is a single point of failure and should not be considered as a solution to guarantee HA.

The solution for `Long latency of the API` may also depend on the nature of the problem. 
For example, if the problem is _users generate OL events on premise and would like to send them to backend run in public cloud_, then being able to buffer the events in own datacenter is cool. On the other hand, if the problem is _lineage backend is in different public cloud region_ which causes delays, then there can be many other (perhaps better) solutions to this problem than the one proposed. I think it's worth giving some examples to help users decide when the proposed solution is appropriate.

Reply:
Thank you for the feedback. Of course this design is not an ultimate solution for the problem and never intend to be, so I rephrased the proposal section to say how it *helps* dealing with the problem. Also mentioned that there could be other possible solutions such as cloud based message broker but it is designed to use as minimal resources as possible. Users will be able to make their own decisions on that.

Reply:
@pawel-big-lebowski does make a good point on calling out logstash. But, the client proxy would provide domain-specific logic on how to process / handle OL events.